src/Tutor.Core/DomainModel/AssessmentEvents/ArrangeTasks/ArrangeTask.cs

```diff
@@ -9,6 +9,12 @@ public class ArrangeTask : AssessmentEvent
         public string Text { get; private set; }
         public List<ArrangeTaskContainer> Containers { get; private set; }
 
+        public override void ValidateInteraction(AssessmentEventInteraction interaction)
+        {
+            if (interaction is not ArrangeTaskInteraction)
+                throw new ArgumentException("Incorrect interaction supplied to Arrange Task with ID " + Id);
```

Comment:
Shouldnt we use Result instead exceptions?

Reply:
I generally based the way this looks like right now entirely off of the methods for evaluating Submissions, which is why it uses ArgumentException. I'm pretty sure I asked about this at an earlier sync (if the current design of Evaluate and Evaluation with exceptions is ok, or if it needs to be refactored in the future), and the conclusion was that it's ok since it's pretty localized.